crates/matrix-sdk/src/send_queue.rs

```diff
@@ -924,6 +924,20 @@ impl QueueStorage {
         let num_initial_dependent_events = dependent_events.len();
 
         let canonicalized_dependent_events = canonicalize_dependent_events(&dependent_events);
+
+        // Get rid of the all non-canonical dependent events.
```

Comment:
Does this fix an existing bug?

Reply:
Yes indeed; didn't add a test for it, because it's a bit far fetched and shouldn't happen in theory that we don't have canonicalized events, and would require lots of setup etc. so little value vs what it's testing.

Reply:
No, happy to follow your instincts.